### ISSUE TITLE
Reverse-Engineer with schemas into directories

### ISSUE THREAD

**Issue by mikethibault** [ErikEJ/EFCorePowerTools#158]
For reverse engineering with multiple schemas, an option to sort classes into schema-named directories would be very helpful.

**Comment by ErikEJ** [comment]
Why have all your tables in a single DbContext?

Why not two projects with a DbContext per schema? (You can do that today)

**Comment by ErikEJ** [comment]
Ping?

**Comment by mikethibault** [comment]
My database has relationships between tables from different schemas.  I'm unaware of a way that Entity Framework can allow joins between contexts without loading everything in memory.  I think it'd be great to mirror this capability of database schemas in an EF Core context.

**Comment by ErikEJ** [comment]
Not something I plan to implement, but a PR for this will be considered

**Comment by Trigun27** [comment]
It will be a cool feature!

**Comment by ErikEJ** [comment]
@Trigun27 PRs accepted 😀

**Comment by Trigun27** [comment]
@ErikEJ Hi. I think this will be a nice feature. Maybe I can try to write the required code.
Any tips and suggestions?

**Comment by ErikEJ** [comment]
Had a closer look at this, looks like an approach to try would be to implement https://github.com/dotnet/efcore/blob/main/src/EFCore.Design/Scaffolding/Internal/RelationalScaffoldingModelFactory.cs#L299 and register this override when building the service provider, if a list of columns to exclude are available, then modify the DatabaseTable model by removing the excluded columns.

we would need a way to pass the list of columns to exclude to this class. 

The list of columns to exclude could be under each Table in the tables list in efpt.config.json.

**Comment by ErikEJ** [comment]
Actually, it does not look like there is any way of doing the in a generic fashion with the current implementation.

This place would be a potential place to do it, but I am not sure you have any schema information at this point: https://github.com/ErikEJ/EFCorePowerTools/blob/master/src/GUI/RevEng.Core/ReverseEngineerRunner.cs#L116

**Comment by ErikEJ** [comment]
Closing as external, must be implemented in EF Core

**Comment by chassq** [comment]
Thanks for re-opening this @ErikEJ !!

**Comment by ErikEJ** [comment]
I will have another look at this, could one of you outline the desired file structure?

Would it be something like:

```
.\Models
    Dbo
        Table1.cs
        Table2.cs
    Otherschema
        Table3.cs
        Table4.cs

```

**Comment by chassq** [comment]
Hey @ErikEJ , See attached our current setup we do manually. Hope this helps. Our schema are named scAccess, for example, in the db. The "sc" prefix is just a naming convention we have to specify the db structure is a schema.

<img width="184" alt="Screenshot 2021-02-27 032203" src="https://user-images.githubusercontent.com/4785384/109381889-faa7b380-78aa-11eb-81f0-14a0b934424a.png">

**Comment by stevie6410** [comment]
> I will have another look at this, could one of you outline the desired file structure?

Hi,
This would be very useful. Is this still something which might be implemented?

**Comment by ErikEJ** [comment]
I am not really able to figure out how to do this without pulling in too much EF Core code, but I added this method recently, so maybe it is within reach: https://github.com/ErikEJ/EFCorePowerTools/blob/master/src/GUI/RevEng.Core/ReverseEngineerScaffolder.cs#L227

A PR to do this would be considered!

**Comment by marcziss** [comment]
Hey Erik, 
Thanks for an awesome toolkit, really makes my life much easier! As far as having multiple folders; I saw that the Handlebars project has support for that
```
services.AddHandlebarsScaffolding(options =>
{
    // Put Models into folders by DB Schema
    options.EnableSchemaFolders = true;
});
```
I assume that utilizing that mechanism won't work here because Reveng does it's own handling of the files and schemas?

**Comment by ErikEJ** [comment]
I guess the implementation could be used as inspiration?

**Comment by ErikEJ** [comment]
Fixed in #981 (Finally) - thanks @eminmesic
